### PR TITLE
rework rpm spec

### PR DESCRIPTION
**Pull request by Jan200101**
Reworks the entire RPM

### CODE REVIEW

File: packages/rpm/polymc.spec [hunk 1]
```diff
@@ -1,13 +1,74 @@
...
-Name: polymc
-Version: 
-Release: 1%{?dist}
-Summary: A custom launcher for Minecraft
-License: GPLv3
-URL: https://polymc.org/
+%global libnbtplusplus_commit       dc72a20b7efd304d12af2025223fad07b4b78464
+%global libnbtplusplus_shortcommit  %(c=%{libnbtplusplus_commit}; echo ${c:0:7})
+%global quazip_commit               c9ef32de19bceb58d236f5c22382698deaec69fd
+%global quazip_shortcommit          %(c=%{quazip_commit}; echo ${c:0:7})
```
Comment (Jan200101) [inline packages/rpm/polymc.spec:5]: Uses the same commits as the repo currently uses

File: packages/rpm/polymc.spec [hunk 2]
```diff
@@ -1,13 +1,74 @@
...
+# zlib/libpng license
+# ---------------------------------------
+# libraries/quazip/quazip/unzip.h
+# libraries/quazip/quazip/zip.h
+#
+
+License:        CC-BY-SA and ASL 2.0 and BSD and Boost and LGPLv2 and LGPLv2+ and LGPLv3+ and GPLv2 and GPLv2+ and GPLv3 and ISC and zlib
+URL:            https://polymc.org
+Source0:        https://github.com/PolyMC/PolyMC/archive/%{version}/%{name}-%{version}.tar.gz
+Source1:        https://github.com/MultiMC/libnbtplusplus/archive/%{libnbtplusplus_commit}/libnbtplusplus-%{libnbtplusplus_shortcommit}.tar.gz
```
Comment (Jan200101) [inline packages/rpm/polymc.spec:61]: it might be worthwhile to fork the libnbtplusplus repo

File: packages/rpm/polymc.spec [hunk 3]
```diff
@@ -19,37 +80,60 @@ Requires: pkgconfig(Qt5)
...
-A custom launcher for Minecraft that allows you to easily manage multiple installations of Minecraft at once
+PolyMC is a free, open source launcher for Minecraft. It allows you to have
+multiple, separate instances of Minecraft (each with their own mods, texture
+packs, saves, etc) and helps you manage them and their associated options with
+a simple interface.
+
 
 %prep
-mkdir -p %{_builddir}/%{name}
-cp -r %{_origdir}/../../* %{_builddir}/%{name}
```
Comment (Jan200101) [inline packages/rpm/polymc.spec:26]: You should generally *never* rely on a local build layout.

**Comment by ghost** [comment]
Instructions on building?

File: packages/rpm/polymc.spec [hunk 4]
```diff
@@ -19,37 +80,60 @@ Requires: pkgconfig(Qt5)
...
+
 
 %changelog
-* Fri Jan 7 2022 getchoo <getchoo at tuta dot io> - 1.0.4
-- Initial polymc spec
+* Sun Jan 09 2022 Jan Drögehoff <sentrycraft123@gmail.com> - 1.0.4-2
+- rework spec
+
+* Fri Jan 7 2022 getchoo <getchoo at tuta dot io> - 1.0.4-1
+- Initial polymc spec
```
Comment (Jan200101) [inline packages/rpm/polymc.spec:139]: updated changelog to follow the format from rpmdev-bumpspec

File: packages/rpm/polymc.spec [hunk 5]
```diff
@@ -19,37 +80,60 @@ Requires: pkgconfig(Qt5)
...
-cd %{_builddir}/%{name}
-cmake \
-  -DLauncher_LAYOUT=lin-system \
-  -DCMAKE_INSTALL_PREFIX=/usr \
-  -DLauncher_LIBRARY_DEST_DIR=%{_lib} \
-  .
+%cmake \
+    -DCMAKE_BUILD_TYPE:STRING="RelWithDebInfo" \
+    -DLauncher_LAYOUT:STRING="lin-system" \
+    -DLauncher_LIBRARY_DEST_DIR:STRING="%{_libdir}/%{name}" \
```
Comment (Jan200101) [inline packages/rpm/polymc.spec:103]: the shared objects build by PolyMC are very PolyMC specific and cannot be relied on by anything else so they are moved into their own directory

**Comment by korewaChino** [comment]
Should be a simple `rpmbuild -bb polymc.spec`.

File: packages/rpm/polymc.spec [hunk 6]
```diff
@@ -1,13 +1,74 @@
...
+# ---------------------------------------
+# libraries/hoedown/
+#
+# zlib/libpng license
+# ---------------------------------------
+# libraries/quazip/quazip/unzip.h
+# libraries/quazip/quazip/zip.h
+#
+
+License:        CC-BY-SA and ASL 2.0 and BSD and Boost and LGPLv2 and LGPLv2+ and LGPLv3+ and GPLv2 and GPLv2+ and GPLv3 and ISC and zlib
```
Comment (Jan200101) [inline packages/rpm/polymc.spec:58]: full license constructed from all vendored dependenices and verified using license-validate

```
$ license-validate -v 'CC-BY-SA and ASL 2.0 and BSD and Boost and LGPLv2 and LGPLv2+ and LGPLv3+ and GPLv2 and GPLv2+ and GPLv3 and ISC and zlib'
Approved license
```

**Review by Jan200101 — COMMENTED** [review]
Tested and compiled on Fedora 35
```
rpmlint polymc.spec polymc-1.0.4-2.fc35.src.rpm
rpmlint: 2.2.0
checks: 32, packages: 2

 1 packages and 1 specfiles checked; 0 errors, 0 warnings, 0 badness; has taken 0.1 s
 ```

I don't know if this will be compatible with non Fedora distros

**Comment by ghost** [comment]
Alright, @Jan200101 if this builds all properly, then I can merge. I'm going to set up a Fedora Jenkins agent later to build this for RPM.

File: packages/rpm/polymc.spec (hunk 2 above)
Comment (ghost) [reply packages/rpm/polymc.spec:61]: not necessary right now, quazip was only forked to fix Windows cross-compilation

Comment (Jan200101) [reply packages/rpm/polymc.spec:61]: alright

**Comment by ghost** [comment]
Looks cool + good to me

**Comment by Jan200101** [comment]
If you are looking to automate Fedora builds I can suggest https://packit.dev/ or maintaining a [copr](https://copr.fedorainfracloud.org)

For the sake of testing remotely I've made a personal copr for it
https://copr.fedorainfracloud.org/coprs/sentry/polymc/

**Pull request merged**